src/outputtype.cpp

```diff
@@ -78,22 +78,30 @@ CTxDestination AddAndGetDestinationForScript(FillableSigningProvider& keystore,
 {
     // Add script to keystore
     keystore.AddCScript(script);
+    ScriptHash sh = ScriptHash(script);
```

Comment:
comment for this whole commit: why is this here? I don't see direct connection with anything. Could this be moved earlier in PR or split out?

Reply:
it was needed for some tests to pass. I've moved it up near the front of the PR.

Reply:
> It's still not clear to me what this is doing.

Nor is it to me. It makes the tests pass. That's. it.

Reply:
Now we know :-) #18067

Reply:
re: https://github.com/bitcoin/bitcoin/pull/16341#discussion_r377810942

Wow, missed this thread. Reviewers: "I don't know what this this doing." Author: "Me either, but it makes tests pass." Reviewers: "ACK" "ACK"

You guys! :heart:

Reply:
There was a bit more back and forth, but yeah... :-(